### PR TITLE
Widen conditions under RESTEasy Reactive Server and RESTEasy Classic Client can work together 

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/resteasy-classic/rest-client/runtime/pom.xml [hunk 1]
```diff
@@ -56,6 +56,10 @@
                     <groupId>org.jboss.spec.javax.ws.rs</groupId>
                     <artifactId>jboss-jaxrs-api_3.0_spec</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.jboss.resteasy</groupId>
+                    <artifactId>resteasy-cdi</artifactId>
+                </exclusion>
```
Comment (gsmet) [inline extensions/resteasy-classic/rest-client/runtime/pom.xml:62]: We are completely positive this is not needed? I wouldn't like breaking perfectly valid use cases to support a corner case.

Comment (geoand) [reply extensions/resteasy-classic/rest-client/runtime/pom.xml:62]: This should never have been here in the first place.

The CDI module provides access to beans that simply don't make sense when you only have the Client and not the Server

**Comment by jamezp** [comment]
> It turns out that the first commit makes the TCK fail (every other testsuite passes). So this probably needs a fix in RESTEasy itself (cc @jamezp )

@geoand What test/TCK is failing? I don't see any being run.

**Comment by geoand** [comment]
@jamezp https://github.com/geoand/quarkus/actions/runs/5975008855/job/16210870660

**Comment by jamezp** [comment]
@geoand It looks like the RESTEasy MicroProfile Client has a dependency on `resteasy-cdi`. I'll have to look to see if that dependency can be removed.

**Comment by geoand** [comment]
Great, thanks!

**Comment by jamezp** [comment]
I don't currently see an easy way to do this. I do have work planned for better CDI support, but that will be for RESTEasy 7.x which will target Jakarta REST 4.0. The only option would be to copy in the classes needed, but I'm not sure that is worth it.

**Comment by stalep** [comment]
Why is the client needing CDI? 
I would still make the argument that using reactive on the server should not force you to use a reactive client.

**Comment by jesperpedersen** [comment]
@jamezp We are working on getting https://github.com/Hyperfoil/Horreum/pull/626 into a committable state, so any help would be appreciated. Currently, RestEasy is blocking this.

**Comment by geoand** [comment]
> Why is the client needing CDI? I would still make the argument that using reactive on the server should not force you to use a reactive client.

Not only this, the beans it provides make no sense in the Client case.

We could perhas just disable the TCK tests that are failing in Quarkus and move this PR ahead...

**Comment by geoand** [comment]
It seems like with change, `ClientHeadersFactoryTest` and `CDIClientHeadersFactoryTest` are failing

**Comment by geoand** [comment]
The exclusion is essentially what I did here and it resulted in the TCK failure I mentioned above

**Comment by geoand** [comment]
Makes sense

**Comment by jamezp** [comment]
I've created https://github.com/resteasy/resteasy-microprofile/issues/203 for this. Hopefully I can look at this today as it seems fairly critical.

**Comment by jamezp** [comment]
@geoand I've created the PR https://github.com/resteasy/resteasy-microprofile/pull/204. Let me know if this works for you.

**Comment by geoand** [comment]
Thanks @jamezp!
That LGTM

**Comment by geoand** [comment]
I don't know at this point. We'll have to see how the Quarkus integration goes

**Comment by stalep** [comment]
Would it be possible to get this into 3.4?

**Comment by geoand** [comment]
@jamezp what's the status of the change you made?

**Comment by jamezp** [comment]
> @jamezp what's the status of the change you made?

@geoand It has been merged. I assume you'll just need me to cut a release. I can get that done this week. Hopefully today or tomorrow.

**Comment by geoand** [comment]
Awesome, thanks!

**Comment by geoand** [comment]
👌

**Comment by stalep** [comment]
@geoand do you have the time to update this pr to use the new resteasy version and see if we can get it into 3.4? :)

**Comment by geoand** [comment]
I've pushed an update to the PR, feel free to try it and let me know how it goes

**Comment by geoand** [comment]
@jamezp I used `2.1.4 ` and now I see some test failures, for example running gives `io.quarkus.restclient.ft.RestClientFallbackTest` results in:

```posh
[INFO] Running io.quarkus.restclient.ft.RestClientFallbackTest
2023-09-07 17:51:45,190 INFO  [io.quarkus] (main) quarkus-rest-client-deployment 999-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 0.356s. Listening on: http://localhost:8081
2023-09-07 17:51:45,192 INFO  [io.quarkus] (main) Profile test activated. 
2023-09-07 17:51:45,192 INFO  [io.quarkus] (main) Installed features: [cdi, rest-client, resteasy, smallrye-context-propagation, smallrye-fault-tolerance, vertx]
2023-09-07 17:51:45,400 ERROR [org.jbo.res.res.i18n] (executor-thread-1) RESTEASY002010: Failed to execute: jakarta.ws.rs.WebApplicationException: HTTP 404 Not Found
	at io.quarkus.restclient.ft.RestClientFallbackTest$TestEndpoint.get(RestClientFallbackTest.java:53)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:356)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:70)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
2023-09-07 17:51:45,430 INFO  [io.quarkus] (main) quarkus-rest-client-deployment stopped in 0.014s
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.969 s <<< FAILURE! -- in io.quarkus.restclient.ft.RestClientFallbackTest
[ERROR] io.quarkus.restclient.ft.RestClientFallbackTest.testFallbackWasUsed -- Time elapsed: 0.209 s <<< ERROR!
jakarta.ws.rs.WebApplicationException: Unknown error, status code 404
	at org.jboss.resteasy.microprofile.client.DefaultResponseExceptionMapper.toThrowable(DefaultResponseExceptionMapper.java:42)
	at org.jboss.resteasy.microprofile.client.ExceptionMapping$HandlerException.mapException(ExceptionMapping.java:60)
	at org.jboss.resteasy.microprofile.client.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:147)
	at jdk.proxy5/jdk.proxy5.$Proxy71.ping(Unknown Source)
	at io.quarkus.restclient.ft.RestClientFallbackTest.testFallbackWasUsed(RestClientFallbackTest.java:35)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.test.QuarkusUnitTest.runExtensionMethod(QuarkusUnitTest.java:499)
	at io.quarkus.test.QuarkusUnitTest.interceptTestMethod(QuarkusUnitTest.java:413)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
```

**Comment by jamezp** [comment]
@geoand That looks weird. Does Quarkus use any kind of CDI interceptor by default? There has not been many code changes https://github.com/resteasy/resteasy-microprofile/compare/2.1.1.Final...2.1.4.Final.

Really just:
- https://github.com/resteasy/resteasy-microprofile/pull/182
- https://github.com/resteasy/resteasy-microprofile/pull/204

**Comment by geoand** [comment]
Let me try with only the bump commit to make sure none of the other changes here are causing that

**Comment by jamezp** [comment]
It must have to do with https://github.com/resteasy/resteasy-microprofile/pull/182. @manovotn do you have any ideas? Is there a change in Quarkus needed too?

**Comment by manovotn** [comment]
> It must have to do with [resteasy/resteasy-microprofile#182](https://github.com/resteasy/resteasy-microprofile/pull/182). @manovotn do you have any ideas? Is there a change in Quarkus needed too?

TBF I didn't think the code I touched is used by Quarkus as it mostly dealt with Portable Extensions (which just don't work in Quarkus, or rather in build time env). But it seems if nothing else, at least `ProxyInvocationHandler` is in the stack trace.
From the top of my head, I have no idea how how Quarkus handles this, would need to take a closer look which I can try to do tomorrow.

**Comment by jamezp** [comment]
It definitely could be https://github.com/resteasy/resteasy-microprofile/pull/182 as I did take some things out when copying those over. I just saw Interceptor in there and it made me wonder if that was it.

**Comment by manovotn** [comment]
> > It must have to do with [resteasy/resteasy-microprofile#182](https://github.com/resteasy/resteasy-microprofile/pull/182). @manovotn do you have any ideas? Is there a change in Quarkus needed too?
> 
> TBF I didn't think the code I touched is used by Quarkus as it mostly dealt with Portable Extensions (which just don't work in Quarkus, or rather in build time env). But it seems if nothing else, at least `ProxyInvocationHandler` is in the stack trace. From the top of my head, I have no idea how how Quarkus handles this, would need to take a closer look which I can try to do tomorrow.

Hmm, it seems Quarkus also made use of that self-constructed interception chain.
Meaning the exception you see here is a result of no interception and hence no `@Fallback` applied.

Since we cannot use interception factory in this scenario, we would have to think of some other way to register those interface impls (currently JDK proxies) as beans so that they get intercepted in a standard way. @geoand isn't RR doesn't something like that already?

Also cc @mkouba @Ladicek for some ideas.

FTR, you could also just revert https://github.com/resteasy/resteasy-microprofile/pull/182 @jamezp as the latest Weld version will work with it again due to some relatively recent changes. However, I find constructing your own interceptor chain to simulate CDI behavior weird at best :)

**Comment by geoand** [comment]
> @geoand isn't RR doesn't something like that already?

Not really, because in RR the impl are generated at build time and they can be registered in a typical Quarkus fashion

**Comment by Ladicek** [comment]
RestClient Reactive generates (during build, before CDI type discovery starts) an implementation of the RestClient interface, which then becomes a plain old managed bean (in the CDI sense, class-based bean). So there are no interface proxies or so.

**Comment by manovotn** [comment]
> > @geoand isn't RR doesn't something like that already?
> 
> Not really, because in RR the impl are generated at build time and they can be registered in a typical Quarkus fashion

That's what I mean - couldn't we use the same logic here?
IIUIC, we can also detect the set of interfaces which have `@RegisterRestClient` and need a backing bean at build time.
Or am I missing something?

**Comment by geoand** [comment]
Looking at https://github.com/quarkusio/quarkus/blob/5a4c0e9a32f8b1376c6f00c5b77d438ea29ae1be/extensions/resteasy-classic/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java#L554 it seems we already do that

**Comment by manovotn** [comment]
> Looking at
> 
> https://github.com/quarkusio/quarkus/blob/5a4c0e9a32f8b1376c6f00c5b77d438ea29ae1be/extensions/resteasy-classic/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java#L554
> it seems we already do that

We are only registering a synth bean based off `RestClientBase` - see [this extension code](https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java#L259-L297), not a full blown, pregenerated bean.

At this point I think the path of least resistance would be to take the hack with custom interceptor chain that was in `resteasy-miprofile` (i.e. the logic previously hosted in [`ProxyInvocationHandler`](https://github.com/resteasy/resteasy-microprofile/pull/182/files#diff-4a626a1618b971eeb4316bec97722c605c52236f6a7ed09d48925d33f82aea31)) and move it directly to Quarkus.
IIUIC, this concerns RE classic only, and Quarkus as such focuses on RR anyway. Thoughts?

**Comment by geoand** [comment]
The reason we are trying to make things easier WRT RESTEasy Classic, is that there are circumstances under which only the Classic REST Classic can be used  - essentially when one wants to use the client to setup testing infrastructure.

> At this point I think the path of least resistance would be to take the hack with custom interceptor chain that was in resteasy-miprofile (i.e. the logic previously hosted in [ProxyInvocationHandler](https://github.com/resteasy/resteasy-microprofile/pull/182/files#diff-4a626a1618b971eeb4316bec97722c605c52236f6a7ed09d48925d33f82aea31)) and move it directly to Quarkus

From my POV, I'm fine with whatever works :)

**Comment by manovotn** [comment]
@geoand @jamezp I've added a commit to this PR which adds `QuarkusProxyInvocationHandler` - a copy of the previous functionality of RE MP `ProxyInvocationHandler` allowing for custom interceptor chains.
All tests in this module pass for me locally:
```
[INFO] Results:
[INFO] 
[INFO] Tests run: 42, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by geoand** [comment]
That's great, thanks a lot @manovotn!

I'll take this PR out of draft in that case.

**Comment by jamezp** [comment]
LGTM

**Comment by geoand** [comment]
@stalep @johnaohara can you give this a try?

It should be good to go otherwise

**Comment by stalep** [comment]
I built Quarkus using this branch and tried to build Horreum from this branch: https://github.com/stalep/repo/tree/quarkus3_nonreactiveclient
Unfortunately I see the same error:
```
[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 4 deployment problems:
[1] Ambiguous dependencies for type jakarta.ws.rs.ext.Providers and qualifiers [@Default]
        - java member: io.quarkus.resteasy.reactive.jackson.runtime.serialisers.FullyFeaturedServerJacksonMessageBodyWriter():providers
        - declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyWriter<java.lang.Object>, org.jboss.resteasy.reactive.server.spi.ServerMessageBodyWriter$AllWriteableMessageBodyWriter, org.jboss.resteasy.reactive.common.types.AllWritea
bleMarker, io.quarkus.resteasy.reactive.jackson.runtime.serialisers.FullyFeaturedServerJacksonMessageBodyWriter, org.jboss.resteasy.reactive.server.spi.ServerMessageBodyWriter<java.lang.Object>, java.lang.Object], qualifiers=[@Default, @Any
], target=io.quarkus.resteasy.reactive.jackson.runtime.serialisers.FullyFeaturedServerJacksonMessageBodyWriter]
        - available beans:
                - PRODUCER METHOD bean [types=[java.lang.Object, jakarta.ws.rs.ext.Providers], qualifiers=[@Default, @Any], target=jakarta.ws.rs.ext.Providers providers(), declaringBean=io.quarkus.resteasy.reactive.server.runtime.QuarkusCon
textProducers]
                - PRODUCER METHOD bean [types=[java.lang.Object, jakarta.ws.rs.ext.Providers], qualifiers=[@Default, @Any], target=jakarta.ws.rs.ext.Providers providers(), declaringBean=org.jboss.resteasy.cdi.ContextProducers]
[2] Ambiguous dependencies for type jakarta.ws.rs.ext.Providers and qualifiers [@Default]
```
@geoand

**Comment by geoand** [comment]
I'll have a look tomorrow

**Comment by geoand** [comment]
I pushed another commit which shiould address the problem

**Comment by geoand** [comment]
I pushed another commit which shiould address the problem

**Comment by stalep** [comment]
I verified that Horreum now starts and that the horreum-backend tests are passing, thanks a lot!

**Review by gastaldi — APPROVED** [review]

**Comment by johnaohara** [comment]
@geoand many thanks!

**Comment by geoand** [comment]
You are welcome!

**Comment by jesperpedersen** [comment]
@gsmet Thanks for considering this for 3.4

**Pull request merged**